src/build/__init__.py

```diff
@@ -233,102 +134,80 @@ def _parse_build_system_table(pyproject_toml: Mapping[str, Any]) -> dict[str, An
     return build_system_table
 
 
+def _wrap_subprocess_runner(runner: RunnerType, env: env.IsolatedEnv) -> RunnerType:
+    def _invoke_wrapped_runner(cmd: Sequence[str], cwd: str | None, extra_environ: Mapping[str, str] | None) -> None:
+        runner(cmd, cwd, {**(env.make_extra_environ() or {}), **(extra_environ or {})})
+
+    return _invoke_wrapped_runner
+
+
 class ProjectBuilder:
     """
     The PEP 517 consumer API.
     """
 
+    _DEFAULT_RUNNER = pyproject_hooks.default_subprocess_runner
+
     def __init__(
-        self,
-        srcdir: PathType,
-        python_executable: str = sys.executable,
-        scripts_dir: str | None = None,
```

Comment:
Was scripts_dir support not needed, I guess?

Reply:
Responsibility for extending the `$PATH` with `scripts_dir` has shifted to the isolated env protocol.  This opens up the possibility of having multiple `script_dir`s and simplifies the relationship between isolated env and builder, with the builder no longer having to manipulate the previously-manipulated environ.